src/main_controller.ts

```diff
@@ -157,6 +157,17 @@ export class MainController implements vscode.Disposable {
 
         this.disposables.push(
             vscode.commands.registerCommand("_getNeovimClient", () => this.client),
+            vscode.commands.registerCommand("vscode-neovim.lua", async (lua) => {
+                if (!lua) {
+                    window.showWarningMessage("No lua code provided");
+                    return;
+                }
+                try {
+                    await this.client.lua(lua);
+                } catch (e) {
+                    logger.error(e instanceof Error ? e.message : e);
```

Comment:
We should add a `vscode.commands.registerCommand` wrapper that all commands use, which could handle errors and show/log them.

Reply:
It's unnecessary. Handling errors or not is the responsibility of the handler.

Reply:
Yes, but at the "top level" in most cases the handling is the same: log it and show a message. That avoids lots of redundant try/catch deeper in the codepath.

When code actually has something useful to do with an error, it of course can use a try/catch.